File: 3436-find-subarray-with-bitwise-or-closest-to-k/find-subarray-with-bitwise-or-closest-to-k.cpp
class Solution {
public:
    int minimumDifference(vector<int>& nums, int k) {
        unordered_set<int>global,curr;
        curr.insert(nums[0]);
        for(int i=1;i<nums.size();i++)
        {   
            for(auto it:curr)
            {
                global.insert(it);
            }

            unordered_set<int>next;
            next.insert(nums[i]);
            for(auto it:curr)
            {
                next.insert(it|nums[i]);
            }
            curr=next;

        }

        for(auto it:curr)
            {
                global.insert(it);
            }
    int m=INT_MAX;
            for(auto it:global)
            {
                m=min(m,abs(k-it));
            }


            return  m;


    }
};

File: 3436-find-subarray-with-bitwise-or-closest-to-k/find-subarray-with-bitwise-or-closest-to-k.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        gl=set()
        curr=set()
        curr.add(nums[0])
        for i in range(1,len(nums)):
            for e in curr:
                gl.add(e)
            
            nxt=set()
            nxt.add(nums[i])
            for e in curr:
                nxt.add(e|nums[i])
            
            curr=nxt
        
        for e in curr:
            gl.add(e)
        ans=float('inf')
        for e in gl:
            ans=min(ans,abs(k-e))
        
        return ans

        